nautobot/core/graphql/__init__.py

```diff
@@ -16,3 +24,24 @@ def convert_field_to_string(field, registry=None):
 def convert_field_to_string(field, registry=None):  # noqa: F811
     """Convert BinaryField to String."""
     return graphene.String()
+
+
+def execute_query(query, variables=None, request=None, user=None):
+    if not request and not user:
+        raise Exception("either request or username should be provided")
+    if not request:
+        request = RequestFactory().request(SERVER_NAME="WebRequestContext")
+        request.id = uuid.uuid4()
+        request.user = user
```

Comment:
This looks very similar to the `nautobot.extras.context_managers.web_request_context` context manager - is there a reason we can't just use that here?